src/private/TitleBar.cpp

```diff
@@ -501,14 +546,33 @@ void TitleBar::onAutoHideClicked()
         return;
     }
 
+    const bool groupedAutoHide = Config::self().flags() & Config::Flag_AutoHideAsTabGroups;
+    const auto currentDw = m_frame->currentDockWidget();
+
     const auto &dockwidgets = m_frame->dockWidgets();
+    if (groupedAutoHide) {
+        SideBarGroupRegister::registerDockWidgetsAsGroup(dockwidgets);
+    }
+
     for (DockWidgetBase *dw : dockwidgets) {
         if (dw->isOverlayed()) {
             // restore
             MainWindowBase *mainWindow = dw->mainWindow();
-            mainWindow->restoreFromSideBar(dw);
+            auto sideBarGroup = groupedAutoHide ? SideBarGroupRegister::groupForDockWidget(dw) : DockWidgetBase::List();
+            if (sideBarGroup.isEmpty()) {
+                mainWindow->restoreFromSideBar(dw);
+            } else {
+                for (auto it = sideBarGroup.rbegin(); it != sideBarGroup.rend(); ++it) {
+                    DockWidgetBase *member = *it;
+                    mainWindow->restoreFromSideBar(member);
+                }
+                dw->setAsCurrentTab();
+            }
+            SideBarGroupRegister::removeDockWidgetFromGroup(dw);
```

Comment:
I think we can remove the whole group here, and not just dw.
I made this change.